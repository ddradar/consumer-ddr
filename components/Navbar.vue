<template>
  <nav class="navbar is-primary" role="navigation" aria-label="main navigation">
    <div class="navbar-brand">
      <NuxtLink class="navbar-item has-text-weight-bold" to="/">
        <img class="images" src="~assets/img/logo.svg" alt="Logo" />
        Consumer DDR
      </NuxtLink>

      <a
        role="button"
        class="navbar-burger"
        aria-label="menu"
        aria-expanded="false"
        data-target="navbarMenu"
      >
        <span aria-hidden="true"></span>
        <span aria-hidden="true"></span>
        <span aria-hidden="true"></span>
      </a>
    </div>

    <div id="navbarMenu" class="navbar-menu">
      <div class="navbar-start">
        <NuxtLink class="navbar-item" to="/songs/">Songs</NuxtLink>
      </div>
      <div class="navbar-end" />
    </div>
  </nav>
</template>
