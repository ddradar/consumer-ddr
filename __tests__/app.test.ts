import { mount, RouterLinkStub } from '@vue/test-utils'
import { describe, expect, test } from 'vitest'

import App from '~~/app.vue'
import Navbar from '~~/components/Navbar.vue'

import { plugins } from './test-utils'

describe('app.vue', () => {
  test('renders correctly', () => {
    // Arrange - Act
    const wrapper = mount(App, {
      global: {
        plugins,
        stubs: { NuxtLink: RouterLinkStub, Navbar, NuxtPage: true }
      }
    })

    // Assert
    expect(wrapper.element).toMatchSnapshot()
  })
})
