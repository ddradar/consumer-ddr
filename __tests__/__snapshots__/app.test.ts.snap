// Vitest Snapshot v1

exports[`app.vue > renders correctly 1`] = `
<div>
  <nav
    aria-label="main navigation"
    class="navbar is-primary"
    role="navigation"
  >
    <div
      class="navbar-brand"
    >
      <a
        class="navbar-item has-text-weight-bold"
      >
        <img
          alt="Logo"
          class="images"
          src="/assets/img/logo.svg"
        />
         Consumer DDR 
      </a>
      <a
        aria-expanded="false"
        aria-label="menu"
        class="navbar-burger"
        data-target="navbarMenu"
        role="button"
      >
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
      </a>
    </div>
    <div
      class="navbar-menu"
      id="navbarMenu"
    >
      <div
        class="navbar-start"
      >
        <a
          class="navbar-item"
        >
          Songs
        </a>
      </div>
      <div
        class="navbar-end"
      />
    </div>
  </nav>
  <section
    class="main-content"
  >
    <nuxt-page-stub />
  </section>
</div>
`;
